starky/src/compressor12/compressor12_pil.rs

```diff
@@ -49,51 +101,340 @@ namespace Compressor(%N);
+        );
+        res.push_str(
+            format!(
+                r#"
+        pol {r0}_B = ({a0} + {a2})  * ({b0} + {b2});
+            "#
+            )
+            .as_str(),
+        );
+        res.push_str(
+            format!(
+                r#"
+        pol {r0}_C = ({a1} + {a2})  * ({b1} + {b2});
+            "#
+            )
+            .as_str(),
+        );
+        res.push_str(
+            format!(
+                r#"
+        pol {r0}_D = {a0} * {b0};
+        "#
+            )
+            .as_str(),
+        );
+        res.push_str(
+            format!(
+                r#"
+        pol {r0}_E = {a1} * {b1};
+        "#
+            )
+            .as_str(),
+        );
+        res.push_str(
+            format!(
+                r#"
+        pol {r0}_F = {a2} * {b2};
+        "#
+            )
+            .as_str(),
+        );
+        res.push_str(
+            format!(
+                r#"
+        pol {r0} = {r0}_C + {r0}_D - {r0}_E - {r0}_F + {c0};
+        "#
+            )
+            .as_str(),
+        );
+        res.push_str(
+            format!(
+                r#"
+        pol {r1} = {r0}_A + {r0}_C - 2*{r0}_E - {r0}_D + {c1};
+        "#
+            )
+            .as_str(),
+        );
+        res.push_str(
+            format!(
+                r#"
+        pol {r2} = {r0}_B - {r0}_D + {r0}_E + {c2};
+        "#
+            )
+            .as_str(),
+        );
+    };
+
+    c_mul_add(
+        "acc1_0", "acc1_1", "acc1_2", "a[0]'", "a[1]'", "a[2]'", "a[3]'", "a[4]'", "a[5]'", "a[9]",
+        "a[10]", "a[11]",
+    );
+    c_mul_add(
+        "acc2_0", "acc2_1", "acc2_2", "acc1_0", "acc1_1", "acc1_2", "a[3]'", "a[4]'", "a[5]'",
+        "a[6]", "a[7]", "a[8]",
+    );
+    c_mul_add(
+        "acc3_0", "acc3_1", "acc3_2", "acc2_0", "acc2_1", "acc2_2", "a[3]'", "a[4]'", "a[5]'",
+        "a[3]", "a[4]", "a[5]",
+    );
+    c_mul_add(
+        "acc4_0", "acc4_1", "acc4_2", "acc3_0", "acc3_1", "acc3_2", "a[3]'", "a[4]'", "a[5]'",
+        "a[0]", "a[1]", "a[2]",
+    );
+
+    // EVPOL4
+    res.push_str(
+        r#"
+    EVPOL4 * (a[6]' - acc4_0 ) = 0;
+    EVPOL4 * (a[7]' - acc4_1 ) = 0;
+    EVPOL4 * (a[8]' - acc4_2 ) = 0;
+
+    // Connection equations
+    {a[0], a[1], a[2], a[3], a[4], a[5], a[6], a[7], a[8], a[9], a[10], a[11]} connect
+        {S[0], S[1], S[2], S[3], S[4], S[5], S[6], S[7], S[8], S[9], S[10], S[11]};
+
+    "#,
+    );
+
     res
 }
+
+#[macro_export]
+macro_rules! c_mul_add {
+    // Note that this is using the `log` function imported above during
+    // `bare_bones`
+    ($($t:tt)*) => (log(&format_args!($($t)*).to_string()))
+}
+
+#[cfg(test)]
+mod test {
+    use crate::compressor12_pil::{render, CompressorPolName::*};
+    use std::fs::File;
+    use std::io::Write;
+    use std::path::Path;
+
+    #[test]
+    fn test_render() {
+        let pil_string = render(5, 5);
+
+        let mut file = File::create(Path::new("./render_pil_rs.pil")).unwrap();
```

Comment:
I would like to create this file in /tmp directory, so it will not affect the .gitignore file.